contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -1801,14 +1801,36 @@ resolve_numeric_typmod_from_exp(Plan *plan, Node *expr, bool *found)
 					precision = tds_default_numeric_precision;
 				}
 
+				if (aggFuncName && strlen(aggFuncName) == 3 &&
+					((strncmp(aggFuncName, "sum", 3) == 0) ||
+					(strncmp(aggFuncName, "avg", 3) == 0)))
+				{
+					/* Handling for fixed length datatype. */
+					if (typmod != -1 && ((*common_utility_plugin_ptr->is_tsql_money_datatype)(aggref->aggtype)))
```

Comment:
Are we sure we do not need to add small money? There is scope of being defensive by adding is_tsql_smallmoney_datatype check even if currently we think it is redundant